modules/gapi/src/backends/streaming/gstreamingbackend.cpp

```diff
@@ -153,8 +153,20 @@ void cv::gimpl::Copy::Actor::run(cv::gimpl::GIslandExecutable::IInput  &in,
         const cv::GRunArgs &in_args = cv::util::get<cv::GRunArgs>(in_msg);
         GAPI_Assert(in_args.size() == 1u);
 
-        cv::GRunArgP out_arg = out.get(0);
-        *cv::util::get<cv::MediaFrame*>(out_arg) = cv::util::get<cv::MediaFrame>(in_args[0]);
+        const auto& in_arg = in_args[0];
+        auto out_arg = out.get(0);
+        using cv::util::get;
+        switch (in_arg.index()) {
+        case cv::GRunArg::index_of<cv::RMat>():
+            *get<cv::RMat*>(out_arg) = get<cv::RMat>(in_arg);
+            break;
+        case cv::GRunArg::index_of<cv::MediaFrame>():
+            *get<cv::MediaFrame*>(out_arg) = get<cv::MediaFrame>(in_arg);
+            break;
+        // FIXME: Add support for remaining types
```

Comment:
Please do this. Handling `Scalar` is quite simple (actually, a copy).

For `OpaqueRef` and `VectorRef`, you can introduce a new method `copy` and implement it in a similar manner to `mov()`. Yes, it will be a copy for those types too, but so far we care most about NOT copying the pixel data.

Reply:
Alternative: remove the template function and replace it with `cv::GMat` / `cv::GFrame` overloads only.

Reply:
Implemented alternative option